index.js

```diff
@@ -24,17 +24,77 @@ module.exports = {
         if (hasRecps(input.content)) {
           return cb(new Error('recps-guard: should not have recps && allowPublic, check your code'))
         }
-        return publish(input.content, cb)
+
+        if (keepOptions) {
+          return publish({...input.content, options: input.options}, cb)
+        } else {
+          return publish(input.content, cb)
+        }
       }
 
       cb(new Error(`recps-guard: public messages of type "${input.type}" not allowed`))
-    })
+    }
+
+    if (ssb.publish) {
+      ssb.publish.hook(publishHook(false))
+
+      ssb.publish.hook = () => {
+        throw new Error('ssb-recps-guard must be the last to hook ssb.publish')
+        // NOTE because of the last hook get run first we need to guarentee
+        // that no other hooks on publish occured after our, otherwise we cannot
+        // guarentee other hooks do not bypass the guard
+      }
+    }
 
-    ssb.publish.hook = () => {
-      throw new Error('ssb-recps-guard must be the last to hook ssb.publish')
-      // NOTE because of the last hook get run first we need to guarentee
-      // that no other hooks on publish occured after our, otherwise we cannot
-      // guarentee other hooks do not bypass the guard
+    if (ssb.tribes && ssb.tribes.publish) {
```

Comment:
I feel like `tribes.publish` should do it's own guard - like no recps => callback with error straight away.
Unlink the other create/publish methods which are used for either/ or.

The other thing is that internally `ssb.tribes.publish` ... uses `ssb.db.create` so it would be covered by that anyway right?

**Recommend:** remove this chunk? Put guard in `tribes.publish`

Reply:
done